pkg/resource/promtail-wiring/reconciler.go

```diff
@@ -67,25 +82,39 @@ func (r *Reconciler) ReconcileDelete(ctx context.Context, lc loggedcluster.Inter
 	logger := log.FromContext(ctx)
 	logger.Info("promtailwiring delete")
 
-	if common.IsWorkloadCluster(lc) {
-		logger.Info("cluster-operator is updating promtailwiring")
-	} else {
-		// Get observability bundle app metadata.
-		appMeta := common.ObservabilityBundleAppMeta(lc)
-
-		var currentApp appv1.App
-		err := r.Client.Get(ctx, types.NamespacedName{Name: appMeta.GetName(), Namespace: appMeta.GetNamespace()}, &currentApp)
-		if err != nil {
-			// Handle case where the app is not found.
-			if apimachineryerrors.IsNotFound(err) {
-				logger.Info("promtailwiring - app not found")
-				return ctrl.Result{}, nil
-			}
-			return ctrl.Result{}, errors.WithStack(err)
+	// Get observability bundle app metadata.
+	appMeta := common.ObservabilityBundleAppMeta(lc)
+
+	var currentApp appv1.App
+	err := r.Client.Get(ctx, types.NamespacedName{Name: appMeta.GetName(), Namespace: appMeta.GetNamespace()}, &currentApp)
+	if err != nil {
+		// Handle case where the app is not found.
+		if apimachineryerrors.IsNotFound(err) {
+			logger.Info("promtailwiring - app not found")
+			return ctrl.Result{}, nil
 		}
+		return ctrl.Result{}, errors.WithStack(err)
+	}
 
-		// Unset user value configmap in the app.
-		if unsetUserConfig(&currentApp, lc) {
+	if lc.IsVintage() {
+		if common.IsWorkloadCluster(lc) {
```

Comment:
I was really hoping we could avoid these conditions by having interfaces :cry: but I don't have a better suggestion at the moment.

Reply:
We could do with interfaces but we'd need to have this setExtraConfig in the interface implémentation :)

Reply:
Well, I'd prefer, as that's supposed to be their role.